staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/decode_test.go

```diff
@@ -45,9 +46,20 @@ func TestDecode(t *testing.T) {
 		want          interface{}
 		assertOnError func(t *testing.T, e error)
 	}{
+		{
+			name: "reject duplicate keys map",
+			in:   hexDecode("a201010102"), // {1:1, 1:2}
```

Comment:
It would be good to have coverage for destination type `map[string]interface{}` with:

- two identical byte string keys
- two identical byte string keys, with one enclosed in an unknown tag
- two identical text string keys
- two identical text string keys, with one enclosed in an unknown tag
- one byte string key and one text string key whose content is identical (only the CBOR major types differ)
- two indefinite-length byte string keys split into chunks differently
- two indefinite-length text string keys split into chunks differently

If there's a case where two CBOR data items are decoded to identical Go map keys without producing a DupMapKeyError, that would be a sneaky issue.